src/swap.ts

```diff
@@ -12,12 +13,40 @@ import { sleep, timeoutPromise, TryParams } from "./util/timeout_promise";
  * It has all the dependencies embedded that are necessary for taking actions on the swap.
  */
 export class Swap {
+  private bitcoinWallet?: BitcoinWallet;
+  private ethereumWallet?: EthereumWallet;
+  private lightningWallet?: LightningWallet;
+
   constructor(
-    private readonly bitcoinWallet: BitcoinWallet,
-    private readonly ethereumWallet: EthereumWallet,
     private readonly cnd: Cnd,
-    readonly self: string
-  ) {}
+    readonly self: string,
+    wallets?: {
```

Comment:
I would create an interface for this and store the `Wallets` object directly inside. Then you can avoid the double optional thing and just default to an empty object inside the swap.
You may also take some inspiration from the `Wallets` class I created for the e2e tests: https://github.com/comit-network/comit-rs/blob/de1f4cae4d57f68fb3e9c2e8986b2b4dabdc4ac4/api_tests/lib/wallets/index.ts

Reply:
It does exactly what you want here:

All wallets are optional but you can access them through an interface that gives you a wallet instance or throws an exception if it doesn't exist.

Reply:
I will do that in a follow-up PR because my OCD would trigger few actions:
- Create a `wallet/` folder in the SDK to contain this new class and existing wallets
- Copy over the `AllWallets` class from e2e
- Use the `AllWallets` class from SDK in e2e, and only extend it for the `initializeForLedger` bit.

Reply:
> Copy over the AllWallets class from e2e

We should carefully consider that but maybe it is the right decision.